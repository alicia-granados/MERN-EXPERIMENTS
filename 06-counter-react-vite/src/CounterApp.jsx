import PropTypes from 'prop-types';

const CounterApp = ({value}) => {
  return (
    <>
      <h1>Counter App</h1>
      <h2> {value} </h2>
    </>
  )
}

export default CounterApp

CounterApp.propTypes ={
    value: PropTypes.number.isRequired,
}
