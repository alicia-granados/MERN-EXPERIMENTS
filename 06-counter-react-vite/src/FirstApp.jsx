const newMessage= 'Fernando';
const newMessage2 ={
    message : 'hola mundo',
    title : 'Fernando'
}
const saludo = () =>{
    return 'Hola como estas'
}
const FirstApp = () => {
    return (
        <>
        <h2> { 1 + 1 } </h2>
        <p>{newMessage} </p>
        <code>{ JSON.stringify(newMessage2)} </code>
        <p>Soy un subtitulo</p>
        <h1>{saludo()}</h1>
        </>
    )
}

export default FirstApp
