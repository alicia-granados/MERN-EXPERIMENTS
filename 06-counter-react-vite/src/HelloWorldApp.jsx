/*export function App () {
    return (<h1>Hola mundo</h1>)
}
export const hola = '123';*/

const HelloWorldApp = () => {
   return (
        <h1>Hola mundo app</h1>
   )
}
 
export default HelloWorldApp
 