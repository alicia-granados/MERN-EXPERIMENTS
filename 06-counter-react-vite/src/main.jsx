import React from 'react'
import { render } from 'react-dom';

const saludo = <h1>Hola mundo</h1>;
const divRoot = document.querySelector('#root');
render ( saludo , divRoot);