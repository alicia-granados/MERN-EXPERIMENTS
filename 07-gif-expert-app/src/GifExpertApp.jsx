const GifExpertApp = () => {
  return (
    <>
        { /*titulo*/ }
        <h1>GifExpertApp</h1>
        { /*input*/ }
        { /*listado de gif - git item*/ }
    </>
  )
}

export default GifExpertApp
