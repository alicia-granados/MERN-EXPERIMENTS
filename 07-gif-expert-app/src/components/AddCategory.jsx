import { useState } from "react";
import PropTypes from 'prop-types';

const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('');
    
    const onInputChange  = ({target}) =>{
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        //console.log(event)
        event.preventDefault();
        if(inputValue.trim().length <=1 ) return;

        //setCategories(categories => [inputValue , ...categories]);
        onNewCategory(inputValue.trim())
        setInputValue('');
    }

    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={ onInputChange}
            />
        </form>
    )
}

export default AddCategory

AddCategory.propTypes={
    onNewCategory: PropTypes.func.isRequired,
}