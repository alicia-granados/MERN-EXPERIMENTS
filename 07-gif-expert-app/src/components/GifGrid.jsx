const GifGrid = ({category}) => {
    const gifs= [1,2,3]
  return (
    <>
      <h3>{category}</h3>
      { 
        gifs.map(gif => (
            <p>{gif}</p>
        ))
      }
    </>
  )
}

export default GifGrid
