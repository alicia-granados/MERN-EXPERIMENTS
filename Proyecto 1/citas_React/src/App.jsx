function App() {
  
  const sumar = () => {
    const numero = 5;
    if(numero > 5){
      console.log("si es mayor")
    }else{
      console.log("no es mayir")
    }
    console.log(2 + 2)
  }
  sumar();

  const edad = 18;

  return (
    <>
      { 1 + 1 }
      {edad >= 18 ? 'eres mayor de edad' : 'no eres mayor de edad'}
      
      <div >
        <h1>{"Hola mundo".toUpperCase()}</h1>
        <p> Un parrafo</p>
      </div>

      <div >
        <h1>{edad}</h1>
        <p> Un parrafo</p>
      </div>
    </>
  )
}

export default App
