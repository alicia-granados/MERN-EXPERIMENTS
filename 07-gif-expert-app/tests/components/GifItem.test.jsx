/** @jest-environment jsdom */
import { render } from "@testing-library/react"
import {GifItem} from "../../src/components";

describe('Pruebas en <GifItem/>', () => {

    const title= 'ateez';
    const url='https://one-punch.com/ateez.jpg';

    test('debe de hacer match con el shapshot',() =>{
        const {container } = render(<GifItem title={title} url = {url}/>)
        expect ( container ).toMatchSnapshot();
    });


})