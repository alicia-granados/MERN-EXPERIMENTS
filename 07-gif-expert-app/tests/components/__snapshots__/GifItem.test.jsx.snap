// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> debe de hacer match con el shapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="ateez"
      src="https://one-punch.com/ateez.jpg"
    />
    <p>
      ateez
    </p>
  </div>
</div>
`;
